File: components/GPU.java
package components; // ← Fará parte do pacote "components", junto com outras peças de computador.
// ↓ Criando a classe GPU (placa de vídeo)
public class GPU {
 protected String nome; // ← Nome da placa (ex.: Radeon RX 550, GeForce RTX 3060, Arc Graphics, etc.)
 protected Integer gb; // ← VRAM em Gigabytes
 protected String tipo; // ← Tipo de VRAM: DDR3, GDDR5, etc.
 protected String conn; // ← Conexão (PCI Express 3.0 x16, etc.) na placa-mãe
 // ↓ Argumentos para criar a classe
 public GPU (String nome, Integer gb, String tipo, String conn) {
  // ↓ NOTA: Os tratadores de exceção abaixo, marcados com if, são personalizados.
  if (nome == null || nome.trim().isEmpty()) {throw new IllegalArgumentException("Não pode ser vazio ou nulo!");}
  if (gb < 0) {throw new IllegalArgumentException("Deve ser maior que zero!");}
  if (tipo == null || tipo.trim().isEmpty()) {throw new IllegalArgumentException("Não pode ser vazio ou nulo!");}
  if (conn == null || conn.trim().isEmpty()) {throw new IllegalArgumentException("Não pode ser vazio ou nulo!");}
  this.nome=nome;
  this.gb=gb;
  this.tipo=tipo;
  this.conn=conn;
 }
 // ↓ Simplesmente digitando o nome do instance dentro do parênteses no "System.out.println()" irá renderizar a string abaixo
 @Override
 public String toString() {
  return "\n::::: TECHPOWERUP GPU DATABASE :::::\nNome: "+nome+"\nVRAM: "+gb+" GB\nTipo de VRAM: "+tipo+"\nConexão: "+conn+"\n";
 }
 // ↓ NOTA: Para manter os atributos da classe como private e para serem visíveis por outras classes, os getters abaixo serão utilizados.
 public String verNome() {return nome;}
 public Integer verVRAM() {return gb;}
 public String verTipoVRAM() {return tipo;}
 public String verConexao() {return conn;}
}

File: components/GPU.py
# ↓ Criando a classe GPU (placa de vídeo)
class GPU:
 lista_tipo=["DDR3","GDDR5"]
 # ↓ Argumentos para criar a classe
 def __init__(self, nome:str, gb:int, tipo:str, conn:str):
 
  # ↓ NOTA: Os tratadores de exceção abaixo, marcados com if, são personalizados.
  if nome is None or nome.strip() == "":
   raise ValueError("Não pode ser vazio ou nulo!")
  if gb <= 0:
   raise ValueError("Deve ser maior que zero!")
  if tipo is None or tipo.strip() == "":
   raise ValueError("Não pode ser vazio ou nulo!")
  if conn is None or conn.strip() == "":
   raise ValueError("Não pode ser vazio ou nulo!")
  if tipo.upper() not in GPU.lista_tipo:
   raise ValueError("Tipo de VRAM não está na lista dos válidos!")
  
  # ↓ NOTA: Para os atributos serem acessíveis à classe filha hcfGPU, os atributos abaixo serão declarados como protegidos. No Python, requer um só underline após o "self." para declarar o atributo como protegido e dois para declarar como privado.
  self._nome=nome # ← Nome da placa (ex.: Radeon RX 550, GeForce RTX 3060, Arc Graphics, etc.)
  self._gb=gb # ← VRAM em Gigabytes
  self._tipo=tipo # ← Tipo de VRAM: DDR3, GDDR5, etc.
  self._conn=conn # ← Conexão (PCI Express 3.0 x16, etc.) na placa-mãe
  
 # ↓ Simplesmente digitando o nome do instance no console, ou dentro do parênteses no "print()" irá renderizar a string abaixo
 def __repr__(self) -> str:
  return f"\n::::: TECHPOWERUP GPU DATABASE :::::\nNome: {self._nome}\nVRAM: {self._gb} GB\nTipo de VRAM: {self._tipo}\nConexão: {self._conn}\n"
  
 # ↓ NOTA: Para manter os atributos da classe como protegidos e para serem visíveis por outras classes, os getters abaixo serão utilizados.
 def verNome(self) -> str:
  return self._nome
 def verVRAM(self) -> int:
  return self._gb
 def verTipoVRAM(self) -> str:
  return self._tipo
 def verConexao(self) -> str:
  return self._conn
